src/Discovery.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\WebLink\GenericLinkProvider;
+use Symfony\Component\WebLink\Link;
+
+final class Discovery
```

Comment:
The name looks broad, can we add some phpdoc to explain what this class is responsible for and who should be using it?